### PR TITLE
fix(issues): a failed surface says so, and clearing one filter clears one

### PR DESCRIPTION
**Pull request by imshashank**
The two things I deferred while reviewing #300, both of which belonged in the
shared code rather than at one call site.

## A surface that could not load said it was empty

Four surfaces checked only whether the request was still pending, so one that
failed fell through to the empty state. My issues said nothing was assigned to
you. A team and a project said they held no issues. A saved view said the same.
None of it true, and none of it offering a way to try again.

The standup board and the sprint already handled this, each carrying its own
copy of the same block, so this adds one shared `LoadFailed` instead of a fifth
and sixth. All six use it now, and the copy says the thing is unread rather than
implying it is empty.

Checked against a stubbed 500 on every one of them, not just the four that were
broken: my issues, a team, a project, a saved view, the standup board and a
sprint each show the retry and none claim to be empty. An end to end test covers
four of the six, and the two that need a project or a view to exist were checked
by hand.

## Clearing the last filter could clear several

`dropLastCondition` sliced the last child off the root group, so when that child
was itself a group it took every condition inside it. `conditionsOf` counts
nested conditions and is what the button is gated on, so the two disagreed about
what a filter holds.

It now descends to the last condition wherever it sits and prunes the group it
empties. Four callers improve together: the `shift+f` hotkey, a project, a team
and a sprint.

Worth saying plainly: nothing in the app builds a nested group today, so this is
latent rather than a bug anyone has hit. I checked before claiming that. It is
still worth fixing in the helper, because the type allows it and the label
promises it.

## Checks

`bun run verify` is green. Both filter tests fail against the old implementation,
which I confirmed by reverting it rather than assuming.

<h3>Greptile Summary</h3>

The PR introduces a shared failed-load state for six issue surfaces and updates nested-filter removal to descend into groups and prune empty branches.
- Adds reusable failed-load messaging and retry controls across issue views.
- Adds component and end-to-end coverage for failed requests.
- Makes dropLastCondition remove one nested condition per invocation.
- Makes unknown organization roles fail closed without throwing.

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

No blocking failure remains.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/web/src/features/issues/load-failed.tsx | Adds the shared failed-request state and retry affordance used by issue surfaces. |
| packages/shared/src/filters/index.ts | Recursively removes the final condition and skips or prunes empty nested groups. |
| packages/shared/src/policy/index.ts | Uses a null-prototype role map and returns no permissions for unrecognized roles. |
| apps/web/e2e/issues.spec.ts | Exercises failed-load rendering across issue surfaces with stubbed server errors. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
  A[Issue surface request] --> B{Request state}
  B -->|Pending| C[Loading skeleton]
  B -->|Failed| D[Shared LoadFailed state]
  D --> E[Try again]
  E --> A
  B -->|Succeeded with no results| F[Empty state]
  B -->|Succeeded with results| G[Issue list or board]
```
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=Greploop%20noveum%2Forbit%20PR%20%23301%3A%20work%20through%20Greptile's%20open%20review%20comments%2C%20then%20keep%20reviewing%20and%20fixing%20until%20it%20comes%20back%20clean%20at%205%2F5%20with%20zero%20unresolved%20comments.%0AStart%20by%20reading%20the%20comments%20off%20the%20PR%20itself.%20On%20GitHub%2C%20use%20paginated%20%60gh%20api%20graphql%60%20to%20query%20%60pullRequest.reviewThreads%60%20with%20each%20thread's%20%60isResolved%60%20value%20and%20inline%20%60comments%60.%20%60gh%20pr%20view%20--comments%60%20only%20includes%20conversation%20comments%2C%20so%20do%20not%20use%20it%20as%20the%20findings%20source.%20They%20are%20not%20listed%20in%20this%20prompt%20on%20purpose%3A%20the%20PR%20is%20current%2C%20a%20pasted%20copy%20would%20not%20be.%20Skip%20anything%20already%20resolved%2C%20and%20if%20you%20judge%20a%20comment%20wrong%2C%20say%20so%20rather%20than%20changing%20code%20to%20satisfy%20it.%0A%0APrefer%20the%20Greptile%20CLI%2C%20which%20reviews%20the%20working%20tree%20with%20no%20push%20and%20no%20CI%20run.%20Fall%20back%20to%20PUSH%20LOOP%20only%20where%20a%20step%20below%20says%20to.%0A1.%20Run%20%60command%20-v%20greptile%60.%20Missing%3A%20go%20to%20PUSH%20LOOP%2C%20and%20pitch%20the%20CLI%20at%20the%20end.%0A2.%20Run%20%60greptile%20whoami%60.%20It%20exits%200%20even%20when%20signed%20out%2C%20so%20read%20the%20output%3A%20if%20it%20says%20%22Not%20signed%20in%22%2C%20ask%20the%20user%20to%20run%20%60greptile%20login%60%20%28it%20opens%20a%20browser%2C%20do%20not%20run%20it%20yourself%29%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%0A3.%20Look%20for%20%60greploop%2FSKILL.md%60%20under%20%60.agents%2Fskills%2F%60%20or%20%60.claude%2Fskills%2F%60%2C%20both%20in%20the%20repo%20root%20and%20in%20the%20user's%20home%20directory.%20Found%3A%20follow%20that%20skill%2C%20never%20pushing%20between%20rounds%2C%20and%20you%20are%20done.%0A4.%20Missing%3A%20run%20%60greptile%20skills%20list%60.%20If%20it%20fails%20as%20an%20unknown%20command%2C%20this%20CLI%20predates%20the%20skill%2C%20so%20tell%20the%20user%20to%20run%20%60greptile%20update%60%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%20Otherwise%20ask%20the%20user%20before%20running%20%60greptile%20skills%20install%20greploop%60%2C%20then%20follow%20the%20skill%20it%20writes.%0A%0APUSH%20LOOP%20%28only%20when%20a%20step%20above%20sends%20you%20here%29%3A%0A%0AGREPLOOP%20ENABLED%3A%20After%20fixing%20all%20issues%2C%20commit%20and%20push%20your%20changes.%20Greptile%20will%20automatically%20re-review%20the%20PR%20via%20webhook.%20Wait%20for%20the%20review%20to%20complete%2C%20then%20read%20the%20new%20review%20comments%20and%20confidence%20score.%20If%20the%20confidence%20score%20is%20below%205%2F5%20or%20there%20are%20unresolved%20comments%2C%20fix%20those%20issues%20and%20push%20again.%20Repeat%20until%20the%20confidence%20score%20is%205%2F5%20with%20zero%20unresolved%20comments.%20Maximum%205%20iterations.%20Do%20NOT%20open%20a%20new%20PR%20%E2%80%94%20keep%20pushing%20to%20the%20same%20branch.%0A%0AIF%20YOU%20ENDED%20UP%20ON%20PUSH%20LOOP%20BECAUSE%20THE%20CLI%20WAS%20MISSING%3A%20once%20the%20loop%20is%20done%2C%20print%20the%20block%20below%20and%20nothing%20else%20about%20the%20CLI.%20Reproduce%20it%20character%20for%20character%20inside%20a%20fenced%20code%20block%2C%20so%20the%20borders%20line%20up.%20Do%20not%20reword%20it%2C%20translate%20it%2C%20summarize%20it%2C%20add%20to%20it%2C%20or%20drop%20the%20box.%20Print%20it%20once%2C%20and%20do%20not%20bring%20the%20CLI%20up%20again%20later%20in%20the%20session.%0A%0A%E2%95%AD%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%95%AE%0A%E2%94%82%20%20You%20can%20run%20greploops%20faster%20locally%20with%20our%20CLI.%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%E2%94%82%0A%E2%94%82%20%20Install%20it%20at%20https%3A%2F%2Fwww.greptile.com%2Fcli%2C%20or%20I%20can%20install%20it%20for%20you.%20%20%E2%94%82%0A%E2%95%B0%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%95%AF%0A%0AIf%20they%20take%20you%20up%20on%20it%2C%20install%20with%20%60npm%20install%20-g%20greptile%60%20%28or%20%60brew%20install%20greptileai%2Ftap%2Fgreptile%60%29%2C%20then%20%60greptile%20skills%20install%20greploop%60.%20Leave%20%60greptile%20login%60%20to%20them%2C%20it%20opens%20a%20browser.&repo=noveum%2Forbit&pr=301&platform=github">Fix All in Greploop</a>

<sub>Reviews (4): Last reviewed commit: ["Merge remote-tracking branch &#39;origin/mai..."](https://github.com/noveum/orbit/commit/2139489bef29c86ac90e2c76c55ed1e661a61ff1) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=52340300)</sub>

**Context used:**

- Rule used - Flag interactive event handler props (`onClick`, `... ([source](https://app.greptile.com/noveum-ai/github/Noveum/orbit/-/custom-context?memory=f997c777-2305-4f26-8379-ab591f935113))

### CODE REVIEW

File: apps/web/e2e/issues.spec.ts [hunk 1]
```diff
@@ -85,3 +85,31 @@ test.fixme('two viewers see issue, board and comment changes without reloading',
...
+  await page.goto(`${BASE}/login`);
+  await page.getByTestId('dev-sign-in-alex@orbit.example').click();
+  await page.waitForURL(`${BASE}/my-issues`);
+
+  for (const [path, testId] of [
+    ['/my-issues', 'retry-my-issues'],
+    ['/team/eng/issues', 'retry-team-issues'],
+    ['/sprints', 'retry-sprint-issues'],
+    ['/standup', 'retry-standup'],
+  ] as const) {
```
[the remark this answers is not in the document]
Comment (imshashank) [reply apps/web/e2e/issues.spec.ts:104]: Fair, added in 2a8fee8. The sweep now covers all six: my issues, a team, a sprint, the standup board, a project and a saved view.

The last two need a project or a view to exist, so the test finds one by navigating rather than hardcoding a slug, and skips if the workspace has none. I checked the guard was not quietly skipping by deleting the project failure branch and confirming the test then fails on retry-project-issues, which it does.

File: packages/shared/src/filters/index.ts [hunk 2]
```diff
@@ -316,7 +316,15 @@ export function removeCondition(group: FilterGroup, property: FilterProperty): F
...
+  const last = group.children.at(-1);
+  if (last === undefined) return group;
+  const withoutLast = group.children.slice(0, -1);
+  if (last.kind === 'condition') return { ...group, children: withoutLast };
+  const trimmed = dropLastCondition(last);
+  return {
+    ...group,
+    children: trimmed.children.length === 0 ? withoutLast : [...withoutLast, trimmed],
+  };
 }
```
[the remark this answers is not in the document]
Comment (imshashank) [reply packages/shared/src/filters/index.ts:329]: Confirmed and fixed in 2a8fee8. I reproduced it before taking the change rather than reading it: with children of [condition, emptyGroup] the condition count was 1 before and 1 after, so the first press cleared nothing.

I used countConditions(last) === 0 rather than children.length === 0, which also covers a branch holding only other empty groups, and each step strictly shortens the child list so it terminates. Three tests cover it: an empty group beside a condition, a group holding only empty groups, and a filter carrying no conditions at all.

My own fix in this PR introduced it, so thank you for the catch.

File: apps/web/tests/features/issues/load-failed.test.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { afterEach, describe, expect, it, mock } from 'bun:test';
+import { cleanup, fireEvent, render, screen } from '@testing-library/react';
+import { LoadFailed } from '../../../src/features/issues/load-failed.tsx';
+
+afterEach(cleanup);
+
+describe('the state a surface shows when its request failed', () => {
+  it('names what could not be read rather than claiming it is empty', () => {
+    render(<LoadFailed subject="this sprint" onRetry={() => undefined} testId="retry-x" />);
```
[the remark this answers is not in the document]
Comment (imshashank) [reply apps/web/tests/features/issues/load-failed.test.tsx]: Changed in 2a8fee8, all three tests now pass a real spy. The third already asserted the click reaches the handler, so the contract was covered, but a button wired to a handler that does nothing is a poor thing to leave sitting in the fixture that documents the component.

File: packages/shared/src/policy/index.ts [hunk 4]
```diff
@@ -85,12 +85,14 @@ export interface Principal {
...
+const NO_PERMISSIONS: readonly Permission[] = [];
+
 export function permissionsFor(role: OrgRole): readonly Permission[] {
-  return PERMISSIONS_BY_ROLE[role];
+  return PERMISSIONS_BY_ROLE[role] ?? NO_PERMISSIONS;
 }
 
 export function can(principal: Principal, permission: Permission): boolean {
-  return PERMISSIONS_BY_ROLE[principal.role].includes(permission);
+  return permissionsFor(principal.role).includes(permission);
```
[the remark this answers is not in the document]
Comment (imshashank) [reply packages/shared/src/policy/index.ts:98]: Right, and my fix was incomplete. Confirmed it before changing anything: `constructor`, `toString` and `hasOwnProperty` came back as functions and `__proto__` as an object, so none of them were undefined, the `??` never fired, and `can` threw on `.includes` exactly as you describe.

Fixed in the commit above by building the table with `Object.create(null)`, so only the four real roles are ever found and nothing is inherited. Those four keys now return an empty permission list and `can` answers false rather than throwing. The real roles are unchanged: admin 25 permissions, member 20, contributor 10, guest 6.

Regression cases cover `constructor`, `toString`, `__proto__`, `hasOwnProperty` and `valueOf`, asserting both `permissionsFor` and `can` for each.
